src/contexts/PostProvider.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useMemo, useReducer } from 'react';
+import { Post } from '../types/Post';
+
+interface Props {
+  children: React.ReactNode,
+}
+
+interface State {
+  selectedPost: Post | null,
+  errorMessage: string,
+  isLoading: boolean,
+}
+
+type Action = {
```

Comment:
use interface prefer. 
Types have unique features but the interface is best for scaling.